models/dataset.go

```diff
@@ -457,17 +457,49 @@ func ValidateDataset(ctx context.Context, dataset *Dataset) error {
 	var invalidFields []string
 	if dataset.URI != "" {
 		dataset.URI = strings.TrimSpace(dataset.URI)
-		datasetURI, err := url.Parse(dataset.URI)
+		_, err := url.Parse(dataset.URI)
 		if err != nil {
 			invalidFields = append(invalidFields, "URI")
 			log.Event(ctx, "error parsing URI", log.ERROR, log.Error(err))
-		} else {
-			if !strings.Contains(datasetURI.Path, "/datasets") || !strings.Contains(datasetURI.Path, dataset.ID) {
-				fmt.Println("hello")
-				invalidFields = append(invalidFields, "URI")
-			}
 		}
 	}
+
+	for i := range dataset.Publications {
+		generalDetails := &dataset.Publications[i]
+		if generalDetails != nil {
+			trimHref(ctx, generalDetails)
```

Comment:
`trimHref()` returns an error so you will need to assign it and do the if != nil check.